scripts/build-upload-docker-images.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```

Comment:
This script would then replace the `make docker docker-push` targets, right?

Reply:
Yes, it will replace these targets to build and upload multi-arch images.

Reply:
@kun-lu20 @Ashmita152 on my macbook I tried running `sudo bash scripts/build-upload-docker-images.sh` (after `brew install buildkit`) and getting the error:
```
error: failed to solve: rpc error: code = Unknown desc = failed to do request: Head http://localhost:5000/v2/baseimg_alpine/blobs/sha256:595b0fe564bb9444ebfe78288079a01ee6d7f666544028d5e96ba610f909ee43: dial tcp 127.0.0.1:5000: connect: connection refused
```

I tried removing the `--push` from the `create-baseimg` target just to explore options to resolve this then added `darwin/amd64` to the platform list but seeing this error:
```
 => ERROR [darwin/amd64 internal] load metadata for docker.io/library/alpine:3.13                                                                                                                                  8.2s
 => [linux/s390x internal] load metadata for docker.io/library/alpine:3.13                                                                                                                                         8.1s
 => [linux/arm64 internal] load metadata for docker.io/library/alpine:3.13                                                                                                                                         8.1s
 => [linux/amd64 internal] load metadata for docker.io/library/alpine:3.13                                                                                                                                         8.1s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                                                      0.0s
------
 > [darwin/amd64 internal] load metadata for docker.io/library/alpine:3.13:
------
error: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: no match for platform in manifest sha256:f51ff2d96627690d62fee79e6eecd9fa87429a38142b5df8a3bfbb26061df7fc: not found
make: *** [create-baseimg] Error 1
```

Any tips on how to resolve or troubleshoot this?

Reply:
I think the cause of the first error is that `docker buildx` could not connect to the local registry container running on port 5000. 
I haven't tried it on macOS, but from the error log, it looks like `alpine:3.13` doesn't support platform `darwin/amd64` yet.

Reply:
Thanks @kun-lu20! I managed to get past the error with:
```
brew install buildkit

# start the local registry
docker run -d -p 5000:5000 --name registry registry:2

# create the buildx container and ensure it can reach the registry container on the host network
docker buildx create --driver-opt=network=host --use
```

then I was able to successfully run:
```
make docker-images-only
```

You can ignore my attempt to add `darwin/amd64`, I forgot that Docker Desktop for mac runs its own linux VM.

Reply:
No problem! I think now you could run `scripts/build-upload-docker-images.sh` successfully as well.